src/Features/CSharp/Portable/Completion/CompletionProviders/ExplicitInterfaceMemberCompletionProvider.CompletionSymbolDisplay.cs

```diff
@@ -0,0 +1,124 @@
+        private static class CompletionSymbolDisplay
+        {
+            public static string ToDisplayString(ISymbol symbol)
+                => symbol switch
+                {
+                    IEventSymbol eventSymbol => ToDisplayString(eventSymbol),
+                    IPropertySymbol propertySymbol => ToDisplayString(propertySymbol),
+                    IMethodSymbol methodSymbol => ToDisplayString(methodSymbol),
+                    _ => "" // This shouldn't happen.
+                };
+
+            private static string ToDisplayString(IEventSymbol symbol)
+                => symbol.Name;
+
+            private static string ToDisplayString(IPropertySymbol symbol)
+            {
+                var builder = new StringBuilder();
+                if (symbol.IsIndexer)
+                {
+                    builder.Append("this");
+                }
+                else
+                {
+                    builder.Append(symbol.Name);
+                }
+
+                if (symbol.Parameters.Length > 0)
+                {
+                    builder.Append('[');
+
+                    AddParameters(symbol.Parameters, builder);
+
+                    builder.Append(']');
+                }
+
+                return builder.ToString();
+            }
+
+            private static string ToDisplayString(IMethodSymbol symbol)
+            {
+                var builder = new StringBuilder();
+                switch (symbol.MethodKind)
+                {
+                    case MethodKind.Ordinary:
+                        builder.Append(symbol.Name);
+                        break;
+                    case MethodKind.UserDefinedOperator:
+                    case MethodKind.BuiltinOperator:
+                        {
+                            builder.Append("operator ");
+                            builder.Append(SyntaxFacts.GetText(SyntaxFacts.GetOperatorKind(symbol.MetadataName)));
+                        }
+                        break;
+                    case MethodKind.Conversion:
+                        {
+
+                            builder.Append("operator ");
+                            AddType(symbol.ReturnType, builder);
+                            break;
+                        }
+                }
+
+                AddTypeArguments(symbol, builder);
+                builder.Append('(');
+                AddParameters(symbol.Parameters, builder);
+                builder.Append(')');
+                return builder.ToString();
+            }
+
+            private static void AddParameters(ImmutableArray<IParameterSymbol> parameters, StringBuilder builder)
+            {
+                var first = true;
+                foreach (var parameter in parameters)
+                {
+                    if (!first)
+                    {
+                        builder.Append(", ");
+                    }
+
+                    first = false;
+                    AddType(parameter.Type, builder);
```

Comment:
any concern about ref-kind?

Reply:
@CyrusNajmabadi Yes. This was a bug. Fixed and added a test.

Reply:
any need to add params?

Reply:
any need to add constraints?  i know it's an explicit interface, so normally no.  but is there that funky case where we need to know if `T?` is unconstrained or not?

Reply:
> any need to add params?

Yes it definitely should be added. It was a working scenario. Fixed in latest commit.

---

> any need to add constraints? i know it's an explicit interface, so normally no. but is there that funky case where we need to know if `T?` is unconstrained or not?

This is a scenario that doesn't work well on latest preview already.

![image](https://user-images.githubusercontent.com/31348972/150420112-844f03f5-643a-4387-8ea2-87b0aa0742f1.png)

The existing behavior is: no `?` is added, and no "default" constraint is added.

I'll probably just add the `?` for this PR and leave any potential improvements to address separately.